framework/src/executioners/PicardSolve.C

```diff
@@ -194,6 +194,7 @@ PicardSolve::solve()
   }
 
   Real pp_scaling = 1.0;
+  std::ostringstream pp_history;
```

Comment:
I would add this below here to output the initial value too:

```c++
if (_picard_custom_pp)
        pp_history << std::setw(2) << " 0 Picard post-processor = "
                 << Console::outputNorm(std::numeric_limits<Real>::max(), *_picard_custom_pp) << "\n";
```

Reply:
The issue is that the initial value could be invalid but because the pp is used for control picard convergence, it will be valid inside the iteration. I did not output the value at the beginning as residual norms purposely.

Reply:
If `direct_pp_value` is true, I agree with you, but if it is false, then this value is valid and is used as `pp_old`. So I guess add `&& !getParam<bool>("direct_pp_value")` to the `if` I put there.

Reply:
This sounds reasonable.

Reply:
Oops, I don't think I should have put the `setw` there.

Reply:
Wait, see the following code:
```
    if (_picard_custom_pp && _picard_it > 0 && !getParam<bool>("direct_pp_value"))
      pp_old = *_picard_custom_pp;
```
pp_old is set only when Picard iteration number greater than 0. So even in that case, we seem should still start outputting after the first picard iteration.

Reply:
Oh good catch, disregard then.